src/renderers/dom/fiber/wrappers/ReactDOMFiberInput.js

```diff
@@ -132,11 +132,8 @@ var ReactDOMInput = {
     node._wrapperState = {
       initialChecked: props.checked != null ? props.checked : props.defaultChecked,
       initialValue: props.value != null ? props.value : defaultValue,
+      controlled: isControlled(props)
     };
-
-    if (__DEV__) {
-      node._wrapperState.controlled = isControlled(props);
-    }
```

Comment:
It was only used for warnings. We intentionally want to minimize state and memory usage associated with controlled components. The goal is to get rid of the entire wrapper object making this a prod dependency makes that harder.